clippy_lints/src/loops/mod.rs

```diff
@@ -577,6 +578,33 @@ declare_clippy_lint! {
     "manual implementation of `Iterator::find`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `for (_, v) in a.iter().enumerate()`
+    ///
+    /// ### Why is this bad?
+    /// The index from `.enumerate()` is immediately dropped.
+    ///
+    /// ### Example
+    /// ```rust
+    /// let v = vec![1, 2, 3, 4];
+    /// for (_, x) in v.iter().enumerate() {
+    ///     print!("{x}")
```

Comment:
`println!` is far more familiar to users than `print!`, it almost seems like a typo for newer users.

(Also the semi-colon)

```suggestion
    ///     println!("{x}");
```

The same with the next example